network/tests/p2p.rs

```diff
@@ -5,65 +5,82 @@
+        test_with_config(&config, 5, false).await?;
+        Ok(())
     }
 
-    pub fn test_with_config(config: &TransportConfig, timeout_secs: u64, success_expected: bool) {
-        async_std::task::block_on(async move {
-            let (transport, peer_id) = {
-                let builder = TransportBuilder::default()
-                    .with_config(config)
-                    .with_timeout(Duration::from_secs(timeout_secs));
-                let tcp = TcpConfig::new().nodelay(true);
-                builder.build(tcp)
-            };
-            let dest: Multiaddr = DEST.parse().unwrap();
-            let behaviour = ping::Behaviour::new(ping::Config::new().with_keep_alive(true));
-            let mut swarm = Swarm::new(transport, behaviour, peer_id);
-            swarm.dial(dest).unwrap();
-            let mut success = false;
-            loop {
-                match swarm.select_next_some().await {
-                    SwarmEvent::NewListenAddr { address, .. } => {
-                        println!("Listening on {:?}", address)
-                    }
-                    SwarmEvent::Behaviour(event) => {
-                        println!("{:?}", event);
-                        event.result.unwrap();
-                        success = true;
-                        break;
-                    }
-                    _ => {
-                        println!("Other");
-                        break;
-                    }
+    pub async fn test_with_config(
+        config: &TransportConfig<'static>,
+        timeout_secs: u64,
+        success_expected: bool,
+    ) -> anyhow::Result<()> {
+        let (transport, peer_id) = {
+            let builder = TransportBuilder::default()
+                .with_config(config)
+                .with_timeout(Duration::from_secs(timeout_secs));
+            builder.build()?
+        };
+        let dest: Multiaddr = DEST.parse().unwrap();
+        let behaviour = ping::Behaviour::new(ping::Config::new().with_keep_alive(true));
+        let mut swarm = SwarmBuilder::new(transport, behaviour, peer_id)
+            // executor has to be explicitly set due to https://github.com/libp2p/rust-libp2p/issues/2173
+            .executor(Box::new(|fut| {
+                tokio::spawn(fut);
+            }))
+            .build();
+        swarm.dial(dest).unwrap();
+        let mut success = false;
+        loop {
+            match swarm.select_next_some().await {
+                SwarmEvent::NewListenAddr { address, .. } => {
+                    // To see this in stdout
+                    // run tests with 'cargo test --release --  --nocapture'
+                    println!("Listening on {:?}", address)
+                }
+                SwarmEvent::Behaviour(event) => {
+                    println!("{:?}", event);
+                    event.result.unwrap();
+                    success = true;
+                    break;
+                }
+                _ => {
```

Comment:
would be good to show/print the value for debugging.

Reply:
Fixed